lhotse/recipes/icmcasr.py

```diff
@@ -15,75 +15,98 @@
+                for sdm_position in SDM_POSITION
+            ]
+        else:
+            raise ValueError(f"Unsupported mic type: {mic}")
+
+        for audio_path, recording_id in zip(audio_paths, recording_ids):
+            # check if audio_path exists, if not, then skip
+            if not audio_path.is_file():
+                # give some warning
+                logging.warning(f"Audio file {audio_path} does not exist - skipping.")
+                continue
+            recordings.append(
+                Recording.from_file(path=audio_path, recording_id=recording_id)
+            )
 
-        audio_path = (section_path / (position + ".wav")).resolve()
-        recording_id = (
-            str(section_path / position)
-            .replace(str(corpus_dir) + "/", "")
-            .replace("/", "-")
-        )
-
-        recordings.append(
-            Recording.from_file(path=audio_path, recording_id=recording_id)
-        )
-
-        with open(text_path) as f:
-            datalines = f.read().splitlines()
-
-        seq = 0
-        for dataline in datalines:
-            if "name" in dataline:
-                speaker = dataline.split('"')[1].strip()
-            elif "xmin =" in dataline:
-                start = float(dataline.split("=")[1].strip())
-            elif "xmax =" in dataline:
-                end = float(dataline.split("=")[1].strip())
-            elif "text" in dataline:
-                text = dataline.split('"')[1].strip()
-                if len(text) > 0:
-                    if float(recordings[-1].duration) < end:
-                        duration = float(recordings[-1].duration) - start
-                    else:
-                        duration = end - start
-                    segment_id = recording_id + "-" + str(seq)
-                    segments.append(
-                        SupervisionSegment(
-                            id=segment_id,
-                            recording_id=recording_id,
-                            start=start,
-                            duration=duration,
-                            channel=0,
-                            language="Chinese",
-                            speaker=speaker,
-                            text=text,
+            with open(text_path) as f:
+                datalines = f.read().splitlines()
+
+            seq = 0
+            for dataline in datalines:
+                if "name" in dataline:
+                    speaker = dataline.split('"')[1].strip()
+                elif "xmin =" in dataline:
+                    start = float(dataline.split("=")[1].strip())
+                elif "xmax =" in dataline:
+                    end = float(dataline.split("=")[1].strip())
+                elif "text" in dataline:
+                    text = dataline.split('"')[1].strip()
+                    if len(text) > 0:
+                        if float(recordings[-1].duration) < end:
+                            duration = float(recordings[-1].duration) - start
```

Comment:
It seems this block is parsing a `textgrid` formatted record. It might be better to use standard package for this --- check out the alimeeting recipe for an example.

Reply:
Thanks. Done.